autosklearn/ensemble_builder.py

```diff
@@ -243,8 +243,12 @@ def run(self):
                 # reduce nbest to reduce memory consumption and try it again
                 if isinstance(self.ensemble_nbest, numbers.Integral) and \
                         self.ensemble_nbest == 1:
-                    self.logger.critical("Memory Exception --"
-                                         " Unable to escape from memory exception")
+                    self.logger.critical(
+                        "Memory Exception -- Unable to further reduce the number of ensemble "
+                        "members -- please restart Auto-sklearn with a higher value for the "
+                        "argument 'ensemble_memory_limit' (current limit is %d MB).",
```

Comment:
Pynisher allows to provide None as memory limit. We can use that as a way to have no memory restriction.

Here the message is casted as integer via the %dm which will cause:
```TypeError: %d format: a number is required, not NoneType```

Probably we should update the API doc to highlight that a None is possible. I guess it is unlikely that a call with None will trigger this crash, but to be on the safe side, should we just use .format() ?

Reply:
That makes sense. I just updated the PR to reflect the possibility to pass a `None` memory limit for the ensemble builder, and also to make sure the error message is printed correctly.